### PR TITLE
Undelete user flow

### PR DESCRIPTION
**Pull request by MrFlashAccount**
#### Tl;dr 
Closes: enso-org/cloud-v2#863
This PR adds the ability to restore user account after deletion(So called soft-deletion)

<details><summary>Demo Presentation</summary>
<p>

#### TODO: video

</p>
</details> 

---

#### Context:
This PR closes enso-org/cloud-v2#863, and it's a frontend part of the issue. 

#### This Change:
1) Change the texts in settings, clarifying that a user can restore his account in 14 days after deletion.
2) Adds a new page (`/restore-account`)
3) If the user logs into the soft-deleted account, the app navigates him to the restore page. 
4) The user can't interact with the dashboard until he restores the account(If the user navigates to the dashboard, the app will redirect him back to the restore page)
5) On the restore page the user can click a "Restore" button to restore his account or log out.
6) If the user restores his account, the app navigates him to the dashboard.

#### Test Plan:
1. Try to delete the account and log back in, expect to see the restore page
2. Try to navigate to protected pages(dashboard, settings, etc)
3. Try to restore the account
4. Try to log out on the restore page
5. (ideally) try to restore the account after 14 days :)

---

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/pages/authentication/RestoreAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import UntrashIcon from 'enso-assets/untrash.svg'
+
+import * as authProvider from '#/providers/AuthProvider'
+import * as backendProvider from '#/providers/BackendProvider'
+import * as textProvider from '#/providers/TextProvider'
+
+import * as ariaComponents from '#/components/AriaComponents'
+import SvgMask from '#/components/SvgMask'
+
+/**
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/pages/authentication/RestoreAccount.tsx:19]: stray comment

File: app/ide-desktop/lib/dashboard/src/providers/AuthProvider.tsx [hunk 2]
```diff
@@ -122,11 +122,24 @@ interface AuthContextType {
...
   readonly resetPassword: (email: string, code: string, password: string) => Promise<boolean>
   readonly signOut: () => Promise<boolean>
+  readonly restoreUser: (backend: Backend) => Promise<boolean>
   /** Session containing the currently authenticated user's authentication information.
    *
    * If the user has not signed in, the session will be `null`. */
   readonly session: UserSession | null
   readonly setUser: React.Dispatch<React.SetStateAction<backendModule.User>>
+  /**
+   * Returns `true` if the user is marked for deletion.
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/providers/AuthProvider.tsx]: avoid using singular verbs (`Returns` -> `Return`):
don't do: `[this] returns a b c`
prefer: `[what this does is] return a b c`

File: app/ide-desktop/lib/dashboard/src/services/Backend.ts [hunk 3]
```diff
@@ -900,6 +901,11 @@ export interface UpdateUserRequestBody {
   username: string | null
 }
 
+/**
+ * HTTP request body for restoring user account.
+ */
+export interface RestoreUserRequestBody {}
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/services/Backend.ts]: unused type

File: app/ide-desktop/lib/dashboard/src/text/english.json [hunk 4]
```diff
@@ -478,6 +479,12 @@
   "saturdayAbbr": "Sa",
   "sundayAbbr": "Su",
 
+  "restoreAccount": "Restore Account",
+  "restoreAccountDescription": "You can restore your account within 14 days of deletion. After 14 days, your account will be permanently deleted.",
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/text/english.json:497]: either use 14 days everywhere, or use 2 weeks everywhere

**Review by somebody1234 — APPROVED** [review]
code review passed
(didn't do QA, might do later)

**Comment by MrFlashAccount** [comment]
@somebody1234 to do qa we have to wait for API to be deployed

**Comment by PabloBuchu** [comment]
1. Is this the same font used everywhere in dashboard?

Screenshot 2024-04-03 at 13 55 04

2. My bad I send you `clear_remove_at` but it follows the same camelCase convention so it should be `clearRemoveAt`. All body and response attributes should be in camelCase (we might have some inconsistency on backend or legacy responses and we should start fixing that)

**Comment by PabloBuchu** [comment]
other than above all looks good. the flow works as expected

**Comment by MrFlashAccount** [comment]
@PabloBuchu 
> Is this the same font used everywhere in dashboard?

I double-checked the font family and yes, they are the same (M PLUS 1)

> My bad I send you clear_remove_at but it follows the same camelCase convention so it should be clearRemoveAt

Sorry, I also missed that! Will update the field on frontend

**Comment by MrFlashAccount** [comment]
@PabloBuchu once you approve, I merge it

**Review by PabloBuchu — APPROVED** [review]
